tests/chainer_tests/functions_tests/math_tests/test_minmax.py

```diff
@@ -341,4 +341,99 @@ def test_pos_neg_duplicate_axis(self):
             self.x.min(axis=(1, -2))
 
 
+@testing.parameterize(*testing.product_dict(
+    [
+        {'function_name': 'argmax', 'function_class': functions.ArgMax},
+        {'function_name': 'argmin', 'function_class': functions.ArgMin},
+    ],
+    [
+        {'dtype': numpy.float16},
+        {'dtype': numpy.float32},
+        {'dtype': numpy.float64},
+    ]
+))
+class TestArgMinMax(unittest.TestCase):
+
+    def setUp(self):
+        self.function = getattr(functions, self.function_name)
+        self.expect = getattr(numpy, self.function_name)
+
+        self.x = numpy.random.uniform(-1, 1, (3, 2, 4)).astype(self.dtype)
+
+    def check_forward(self, x_data, axis=None):
+        x = chainer.Variable(x_data)
+        y = self.function(x, axis=axis)
+        self.assertEqual(y.data.dtype, numpy.int32)
+        y_expect = self.expect(self.x, axis=axis)
+        self.assertEqual(y.data.shape, y_expect.shape)
+        gradient_check.assert_allclose(y_expect, y.data)
+
+    def test_forward_cpu(self):
+        self.check_forward(self.x)
+
+    def test_forward_axis_cpu(self):
+        for i in range(self.x.ndim):
+            self.check_forward(self.x, axis=i)
+
+    def test_forward_negative_axis_cpu(self):
+        self.check_forward(self.x, axis=-1)
```

Comment:
How about testing all valid negative axes, not only -1? The same thing may hold for other negative axis tests below.